src/structures.lisp

```diff
@@ -0,0 +1,1060 @@
+;;; -*- mode:lisp; coding:utf-8 -*-
+
+;;; Tiny DEFSTRUCT for JSCL  @vkm
+;;; Released: :named :type :include :initial-offset
+;;;           :constructor with BOA
+;;;           slot default type read-only syntax
+;;;
+;;; JSCL is free software: you can redistribute it and/or
+;;; modify it under the terms of the GNU General Public License as
+;;; published by the Free Software Foundation, either version 3 of the
+;;; License, or (at your option) any later version.
+;;;
+;;; JSCL is distributed in the hope that it will be useful, but
+;;; WITHOUT ANY WARRANTY; without even the implied warranty of
+;;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+;;; General Public License for more details.
+;;;
+;;; You should have received a copy of the GNU General Public License
+;;; along with JSCL.  If not, see <http://www.gnu.org/licenses/>.
+
+(/debug "loading structures.lisp!")
+
+#-jscl
+(setq *structures* (make-hash-table :test #'equal))
+#+jscl
+(setq *structures* (make-hash-table :test #'eql))
+
+;;; list length 1
+(defun singleton-p (list)
```

Comment:
Should we move all those helpers to some `src/utils.lisp` so they can use be across jscl?

Reply:
>Should we move all those helpers to some src/utils.lisp so they can use be across jscl?

Yes, I agree.
But step by step.
On the next iteration.